autogpt_platform/backend/backend/blocks/github/triggers.py

```diff
@@ -145,8 +148,18 @@ def __init__(self):
             ],
         )
 
-    def run(self, input_data: Input, **kwargs) -> BlockOutput:  # type: ignore
+    def run(  # type: ignore
+        self, input_data: Input, *, credentials: GithubCredentials, **kwargs
+    ) -> BlockOutput:
+        api = get_api(credentials, convert_urls=False)
         yield from super().run(input_data, **kwargs)
+        if input_data.payload["pull_request"]["patch_url"]:
+            response = api.get(
+                input_data.payload["pull_request"]["patch_url"], allow_redirects=True
+            )
+            yield "patch", response.text
+        else:
+            yield "patch", ""
```

Comment:
thoughts on a checkbox?

Reply:
but also for me they're pretty quick and getting the diff on a pr when you're notified of it seems pretty reasonable to me

Reply:
Checkbox could be good. Still, there are multiple ways to represent a diff, so I think it deserves its own block rather than stuffing *a version* of it in this block and `GitHubReadPullRequestBlock`.

Reply:
removed. I still think this is a reasonable behavior to want but don't think its a good experience without dynamic output pins